### PR TITLE
RBAC: Check `plugins:install` globally

### PR DESCRIPTION
**Pull request by gamab**
**What is this feature?**

Plugin installation has impacts on all organizations. Currently we perform an access control check in the organization the user is connected to. The check should be done globally.

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana-enterprise/issues/6055

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/api/api.go [hunk 1]
```diff
@@ -394,8 +394,8 @@ func (hs *HTTPServer) registerRoutes() {
 
 		if hs.Cfg.PluginAdminEnabled && (hs.Features.IsEnabledGlobally(featuremgmt.FlagManagedPluginsInstall) || !hs.Cfg.PluginAdminExternalManageEnabled) {
 			apiRoute.Group("/plugins", func(pluginRoute routing.RouteRegister) {
-				pluginRoute.Post("/:pluginId/install", authorize(ac.EvalPermission(pluginaccesscontrol.ActionInstall)), routing.Wrap(hs.InstallPlugin))
-				pluginRoute.Post("/:pluginId/uninstall", authorize(ac.EvalPermission(pluginaccesscontrol.ActionInstall)), routing.Wrap(hs.UninstallPlugin))
+				pluginRoute.Post("/:pluginId/install", authorizeInOrg(ac.UseGlobalOrSingleOrg(hs.Cfg), ac.EvalPermission(pluginaccesscontrol.ActionInstall)), routing.Wrap(hs.InstallPlugin))
+				pluginRoute.Post("/:pluginId/uninstall", authorizeInOrg(ac.UseGlobalOrSingleOrg(hs.Cfg), ac.EvalPermission(pluginaccesscontrol.ActionInstall)), routing.Wrap(hs.UninstallPlugin))
```
Comment (dmihai) [inline pkg/api/api.go:398]: Normally these changes should be covered by automated tests. Do you think it's doable/easy to add some tests?

Comment (gamab) [reply pkg/api/api.go:398]: Yep, I was doing exactly this :)

Comment (gamab) [reply pkg/api/api.go:398]: I'm still missing some cases to test btw :)

File: pkg/api/plugins_test.go [hunk 2]
```diff
@@ -67,15 +70,21 @@ func Test_PluginsInstallAndUninstall(t *testing.T) {
 			hs.Cfg = setting.NewCfg()
 			hs.Cfg.PluginAdminEnabled = tc.pluginAdminEnabled
 			hs.Cfg.PluginAdminExternalManageEnabled = tc.pluginAdminExternalManageEnabled
+			hs.Cfg.RBACSingleOrganization = true
```
Comment (gamab) [inline pkg/api/plugins_test.go:73]: wip

**Review by IevaVasiljeva — APPROVED** [review]
LGTM! I left some suggestions, but overall I think the change makes sense. I can't think of a nicer way to do it with the information that we currently have from Grafana's config.

File: pkg/setting/setting.go [hunk 3]
```diff
@@ -1694,6 +1696,7 @@ func readAccessControlSettings(iniFile *ini.File, cfg *Cfg) {
 	cfg.RBACPermissionCache = rbac.Key("permission_cache").MustBool(true)
 	cfg.RBACPermissionValidationEnabled = rbac.Key("permission_validation_enabled").MustBool(false)
 	cfg.RBACResetBasicRoles = rbac.Key("reset_basic_roles").MustBool(false)
+	cfg.RBACSingleOrganization = rbac.Key("single_organization").MustBool(false)
```
[the remark this answers is not in the document]
Comment (gamab) [reply pkg/setting/setting.go:1699]: Yes, I agree. I actually have a branch with that change ready :)
Do you think I should add a comment to the config option stating that it's subject to change?

File: pkg/services/pluginsintegration/pluginaccesscontrol/accesscontrol.go [hunk 4]
```diff
@@ -59,7 +59,7 @@ func DeclareRBACRoles(service ac.Service, cfg *setting.Cfg) error {
 		Role: ac.RoleDTO{
 			Name:        ac.FixedRolePrefix + "plugins:maintainer",
 			DisplayName: "Plugin Maintainer",
-			Description: "Install, uninstall plugins",
+			Description: "Install, uninstall plugins. Needs to be assigned globally.",
```
[the remark this answers is not in the document]
Comment (gamab) [reply pkg/services/pluginsintegration/pluginaccesscontrol/accesscontrol.go:62]: Yes, good call.

**Pull request merged**